.meta/dietpi-survey_report

```diff
@@ -8,14 +8,7 @@
 	#-----------------------------------------------------------
 	#	NB: PLEASE ENSURE HW_MODEL INDEX ENTRIES MATCH : PREP, dietpi-obtain_hw_model, dietpi-survey_results,
 	#	NBB: DO NOT REORDER INDEX's. These are now fixed and will never change (due to survey results etc)
-	G_HW_MODEL_MAX=200
 	aHW_NAME=()
-	for (( i=0; i<=$G_HW_MODEL_MAX; i++ ))
```

Comment:
The reason I did this is:
- If we add a new HW_MODEL and forget to update this script, it will fail. At least with the HW_MODEL index displayed, we can see it needs to be updated.
- Its also good general practice to initilize all expected array indexs to either NULL or some value, to prevent a misfire memory read on a non-existent array index.

^^

Mostly due to my C++ background.

Reply:
@Fourdee 
Jep, I solved this with `${aHW_NAME[$i]:-$i}` when printing to html page. So regardless how much and with what index a wrong HW_MODEL was cached, it always is printed rawly without the need to pre-create the array.

This also reduces array loops later:
```
--	for (( i=0; i<=$G_HW_MODEL_MAX; i++ ))
++	for i in ${!aBENCH_INDEX[@]}
++	do
```

Did the same for the other translation arrays, always adding untranslated ID, if it's missing in translation array.

Reply:
@MichaIng 

Legend 👍